src/Uno.UI.Runtime.Skia.Win32/ApplicationMode/DataTransfer/DragDrop/Win32DragDropExtension.DropTarget.cs

```diff
@@ -0,0 +1,160 @@
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Windows.ApplicationModel.DataTransfer;
+using Windows.ApplicationModel.DataTransfer.DragDrop.Core;
+using Windows.Foundation;
+using Windows.Win32;
+using Windows.Win32.Foundation;
+using Windows.Win32.System.Com;
+using Windows.Win32.System.Ole;
+using Windows.Win32.System.SystemServices;
+using Uno.Disposables;
+using Uno.Foundation.Logging;
+using IDataObject = Windows.Win32.System.Com.IDataObject;
+
+namespace Uno.UI.Runtime.Skia.Win32;
+
+// IDropTarget implementation for handling drag-and-drop operations
+internal partial class Win32DragDropExtension
+{
+	unsafe HRESULT IDropTarget.Interface.DragEnter(IDataObject* dataObject, MODIFIERKEYS_FLAGS grfKeyState, POINTL pt, DROPEFFECT* pdwEffect)
+	{
+		Debug.Assert(_manager is not null && _coreDragDropManager is not null);
+
+		IEnumFORMATETC* enumFormatEtc;
+		var hResult = dataObject->EnumFormatEtc((uint)DATADIR.DATADIR_GET, &enumFormatEtc);
+		if (hResult.Failed)
+		{
+			this.LogError()?.Error($"{nameof(IDataObject.EnumFormatEtc)} failed: {Win32Helper.GetErrorMessage(hResult)}");
+			return HRESULT.E_UNEXPECTED;
+		}
+
+		using var enumFormatDisposable = new DisposableStruct<IntPtr>(static p => ((IEnumFORMATETC*)p)->Release(), (IntPtr)enumFormatEtc);
+
+		enumFormatEtc->Reset();
+		const int formatBufferLength = 100;
+		var formatBuffer = stackalloc FORMATETC[formatBufferLength];
+		uint fetchedFormatCount;
+		hResult = enumFormatEtc->Next(formatBufferLength, formatBuffer, &fetchedFormatCount);
+		if (hResult.Failed)
+		{
+			this.LogError()?.Error($"{nameof(PInvoke.RegisterDragDrop)} failed: {Win32Helper.GetErrorMessage(hResult)}");
+			return HRESULT.E_UNEXPECTED;
+		}
+
+		var position = new System.Drawing.Point(pt.x, pt.y);
+
+		var success = PInvoke.ScreenToClient(_hwnd, ref position);
+		if (!success) { this.LogError()?.Error($"{nameof(PInvoke.ScreenToClient)} failed: {Win32Helper.GetErrorMessage()}"); }
+		var scaledPosition = GetScaledPosition(position.X, position.Y);
+
+		var src = new DragEventSource(scaledPosition, grfKeyState);
+
+		var formats = new Span<FORMATETC>(formatBuffer, (int)fetchedFormatCount);
+		if (this.Log().IsEnabled(LogLevel.Trace))
+		{
+			var log = $"{nameof(IDropTarget.Interface.DragEnter)} @ {position}, formats: ";
+			foreach (var format in formats)
+			{
+				log += (CLIPBOARD_FORMAT)format.cfFormat + " ";
+			}
+			this.Log().Trace(log);
+		}
+
+		var package = new DataPackage();
+
+		var formatEtcList = formats.ToArray();
+		var formatList =
+			formatEtcList
+			.Where(static formatetc =>
+			{
+				if (!Enum.IsDefined((CLIPBOARD_FORMAT)formatetc.cfFormat))
+				{
+					return false;
+				}
+				if (formatetc.tymed != (uint)TYMED.TYMED_HGLOBAL)
+				{
+					typeof(Win32DragDropExtension).LogError()?.Error($"{nameof(IDropTarget.Interface.DragEnter)} found {Enum.GetName((CLIPBOARD_FORMAT)formatetc.cfFormat)}, but {nameof(TYMED)} is not {nameof(TYMED.TYMED_HGLOBAL)}");
```

Comment:
Could the message explain some more about what happens and what can be done about it?

Reply:
This should not happen unless we add new formats that are not correctly supported yet.